Oleksandr.Svidovyi/Task_6/src/main/java/com/hillel/task_6/controller/ApiAccountController.java

```diff
@@ -0,0 +1,144 @@
+package com.hillel.task_6.controller;
+
+import com.hillel.task_6.model.Account;
+import com.hillel.task_6.model.Client;
+import com.hillel.task_6.model.request_model.TransferRequestModel;
+import com.hillel.task_6.model.request_model.UpdateAccountRequestModel;
+import com.hillel.task_6.service.AccountService;
+import com.hillel.task_6.service.ClientService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/clients/{id}/accounts")
+public class ApiAccountController {
+
+    @Autowired
+    private ClientService clientService;
+
+    @Autowired
+    @Qualifier("testAccountServiceImpl")
+    private AccountService accountService;
+
+
+    @GetMapping(path = "/list")
+    public ResponseEntity<List<Account>> getClientAccounts(@PathVariable int id) {
+        Client client = clientService.getClient(id);
+
+        if (client == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        List<Account> accounts = clientService.getClient(id).getAccounts();
```

Comment:
What if clientService.getClient(id) is null?

Create method clientService.getAccountsByClientId(int clientId)

Reply:
Since we fetch clientId from URL wich is formed from concrete client, I asume client already exists.

Method clientService.getAccountsByClientId(int clientId) was added.

Reply:
Method clientService.getAccountsByClientId(int clientId) was added.